client/me/purchases/manage-purchase/index.jsx

```diff
@@ -500,6 +541,203 @@ class ManagePurchase extends Component {
 		return null;
 	}
 
+	renderRemoveSubscriptionWarningDialog( site, purchase ) {
+		if ( this.state.showRemoveSubscriptionWarningDialog ) {
+			const { hasCustomPrimaryDomain, primaryDomain } = this.props;
+			const primaryDomainName = hasCustomDomain && primaryDomain ? primaryDomain.name : '';
+			const isPlanRefundable = isRefundable( purchase );
+			const actionFunction = isPlanRefundable
+				? this.goToCancelLink
+				: this.showPreCancellationModalDialog;
+
+			let customDomain = false;
```

Comment:
This can be just `const customDomain = hasCustomPrimaryDomain && primaryDomainName;`. I'm wondering if we really need this though. See the commet below regarding the `hasDomain` prop.